src/python/bot/fuzzers/libFuzzer/engine.py

```diff
@@ -372,23 +370,62 @@ def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
     runner = libfuzzer.get_runner(target_path)
     libfuzzer.set_sanitizer_options(target_path)
     merge_tmp_dir = self._create_temp_corpus_dir('merge-workdir')
-
-    merge_result = runner.merge(
-        [output_dir] + input_dirs,
-        merge_timeout=max_time,
-        tmp_dir=merge_tmp_dir,
-        additional_args=arguments,
-        artifact_prefix=reproducers_dir)
-
-    if merge_result.timed_out:
-      raise MergeError('Merging new testcases timed out')
-
-    if merge_result.return_code != 0:
-      raise MergeError('Merging new testcases failed')
+    merge_control_file = os.path.join(merge_tmp_dir, 'MCF')
+
+    # Make sure the merge control file does not exist at this point.
+    shell.remove_file(merge_control_file)
+
+    additional_args = arguments + [
+        '%s%s' % (constants.MERGE_CONTROL_FILE_ARGUMENT, merge_control_file)
+    ]
+    merge_stats = {}
+
+    # Two step merge process to obtain accurate stats for the new corpus units.
+    # See https://reviews.llvm.org/D66107 for a more detailed description.
+    def _merge_step(corpus_directories, step):
+      """Execute the fuzz target, check the result, record the stats."""
+      result = runner.merge(
+          corpus_directories,
+          merge_timeout=max_time,
+          tmp_dir=merge_tmp_dir,
+          additional_args=additional_args,
+          artifact_prefix=reproducers_dir)
+
+      if result.timed_out:
+        raise MergeError('Merging new testcases (step %d) timed out' % step)
```

Comment:
nit: I think we generally end the error messages with a period (though we're not entirely consistent and I don't really think it's that helpful to have). Here and 398.

Reply:
+1, but I've reverted to the original messages here, i.e. leaving those without trailing periods